core/bin/zksync_api/src/api_server/rest/v01/api_impl.rs

```diff
@@ -65,27 +66,57 @@ impl ApiV01 {
                 .expect("TickerConfig::liquidity_volume must be positive"),
         );
 
+        let ticker_config = zksync_config::configs::TickerConfig::from_env();
+        let unconditionally_valid_addresses = ticker_config.unconditionally_valid_tokens;
+
         let mut storage = self_.access_storage().await?;
-        let mut tokens = storage
+        let tokens = storage
             .tokens_schema()
-            .load_tokens_by_market_volume(liquidity_volume)
+            .load_tokens()
             .await
             .map_err(Self::db_error)?;
 
-        // Add ETH for tokens allowed for fee
-        // Different APIs have different views on how to represent ETH in their system.
-        // But ETH is always allowed to pay fee, and in all cases it should be on the list.
-
-        if tokens.get(&TokenId(0)).is_none() {
-            let eth = Token::new(TokenId(0), Default::default(), "ETH", 18, TokenKind::ERC20);
-            tokens.insert(eth.id, eth);
-        }
+        // I know this seems like a very inefficient way of doing it, when we could just make a left join in sql and in fact I've got that code stashed for two reasons:
+        // 1. the storage manager should not really have any knowledge of the unconditionally valid tokens
+        // 2. and more importantly, I couldn't make it work, due to limitations of myself and/or sqlx (the 'token.address IN ()' seems to be problematic when not dealing with hardcoded values)
+        // We should optimise this at some point tho  (#tech-dept)
+        let mut unconditionally_valid_tokens: Vec<Token> = Vec::new();
+        let tokens_to_filter: HashMap<TokenId, Token> = tokens
+            .into_iter()
+            .filter(|(_, token)| {
+                if unconditionally_valid_addresses.contains(&token.address) {
+                    unconditionally_valid_tokens.push(token.clone());
```

Comment:
Why do we add a token to `unconditionally_valid_addresses` if it already contains it?